model-optimizer/extensions/ops/If.py

```diff
@@ -0,0 +1,319 @@
+
+    @staticmethod
+    def connect_body_output(if_node: Node, condition: bool, if_output_port_idx: int, internal_result: Node):
+        """
+        Update the output port map to connect the body Result node with the specified output port
+
+        :param if_node: the If node
+        :param condition: the boolean defining a condition (then/else) graph to add connect the body
+        :param if_output_port_idx: the output port index to connect
+        :param internal_result: the body Result node to connect
+        :return: None
+        """
+        assert if_node.soft_get('op') == 'If'
+        assert internal_result.soft_get('op') == 'Result'
+        sub_graph = if_node.then_graph if condition else if_node.else_graph
+        assert internal_result.id in sub_graph
+        internal_result['output_id'] = if_output_port_idx
+
+    @staticmethod
+    def update_body_parameters_type(if_node: Node, condition: bool):
+        """
+        Update the data type for If body Parameter nodes based on data type of the outer graph nodes producing data
+        for them.
+
+        :param if_node: The If node
+        :param condition: the boolean defining a condition (then/else) graph
+        :return: None
+        """
+        assert if_node.soft_get('type') == 'If'
+
+        subgraph = if_node.then_graph if condition else if_node.else_graph
+        for node in subgraph.get_op_nodes():
+            if node.has('input_id'):
+                assert node.soft_get('type') == 'Parameter'
+                input_port_id = node['input_id']
+                input_type = if_node.in_port(input_port_id).get_data_type()
+                node.data_type = input_type
+                log.debug('Updated data type for the body node with name "{}" with value {}'
+                          .format(node.name, node.data_type))
+
+    @staticmethod
+    def updated_body_parameters_shape(if_node: Node, condition: bool):
+        """
+        Update shape for If body parameters.
+
+        :param if_node: The If node
+        :param condition: the boolean defining a condition (then/else) graph to add connect the body
+        :return: None
+        """
+        subgraph = if_node.then_graph if condition else if_node.else_graph
+        for node in subgraph.get_op_nodes():
+            if node.has('input_id'):
+                assert node.soft_get('type') == 'Parameter'
+                input_port_id = node['input_id']
+                input_shape = if_node.in_port(input_port_id).get_connection().get_source().data.get_shape()
+                if node.soft_get('shape', None) is None:
+                    node['shape'] = None
+                node.shape = input_shape.copy()
+                log.debug('Updated shape for the body node with name "{}" with value {}'
+                          .format(node.soft_get('name', node.soft_get('id')), node.shape))
+
+    @staticmethod
+    def updated_if_output_ports_shape(if_node: Node):
```

Comment:
rename this function and `updated_if_output_ports_type`. Should it be `update_` instead of `updated_`?